hg/hg_test.go

```diff
@@ -27,6 +27,11 @@ var _ = Describe("Hg", func() {
 			Expect(unionOfPaths(repo.IncludePaths)).To(Equal("file('re:/path/1')|file('re:/path/2')|file('re:/path/3')"))
 		})
 
+		It("produces the correct branch fragment", func() {
+			Expect(repo.makeBranchQueryFragment()).To(Equal("literal:re:a_branch"))
+			Expect(emptyRepo.makeBranchQueryFragment()).To(Equal(""))
```

Comment:
If the user didn't explicitly specify a branch, we should probably assume `default`, not an empty string.

An empty unquoted string produces a parse error:
```sh
$ hg log --rev "branch()"
hg: parse error: missing argument
```

And an empty quoted string returns results from the currently checked out branch:
```sh
$ hg checkout bar-branch
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
$ hg log --rev "branch('')"
changeset:   1:972cc008a6d4
branch:      bar-branch
user:        Foo
date:        Wed Nov 16 09:27:05 2016 +0100
summary:     bar

$ hg checkout foo-branch
2 files updated, 0 files merged, 0 files removed, 0 files unresolved
$ hg log --rev "branch('')"
changeset:   2:a649559327f9
branch:      foo-branch
tag:         tip
user:        Foo
date:        Wed Nov 16 09:27:26 2016 +0100
summary:     foo
```